r/R/dplyr-datetime-helpers.R

```diff
@@ -179,7 +179,24 @@ build_formats <- function(orders) {
     orders <- unique(c(orders1, orders2))
   }
 
-  supported_orders <- c("ymd", "ydm", "mdy", "myd", "dmy", "dym")
+  ymd_orders <- c("ymd", "ydm", "mdy", "myd", "dmy", "dym")
+  ymd_hms_orders <- c(
+    "ymd_HMS", "ymd_HM", "ymd_H", "dmy_HMS", "dmy_HM", "dmy_H", "mdy_HMS",
+    "mdy_HM", "mdy_H", "ydm_HMS", "ydm_HM", "ydm_H"
+  )
+  # support "%I" hour formats
+  ymd_ims_orders <- gsub("H", "I", ymd_hms_orders)
```

Comment:
Is this ok to do? I thought `H` was 24-hour hours and `I` was 12? Do we have a test for using H with 24-hour style hours?

Reply:
Yes, this is used to add to the whitelist of supported orders, based off an already hardcoded list containing `"H"`. See my other [comment](https://github.com/apache/arrow/pull/13196#discussion_r880193852). We could always remove this code and figure out a different way to message the user. 
The whitelist approach was introduced as response to [this bit of feedback](https://github.com/apache/arrow/pull/12589#discussion_r861780324).

Reply:
You mean, tests for using `"I"` with 24h?

Reply:
Added a [test](https://github.com/dragosmg/arrow/blob/4846f806328c3f63f9d3aaca24ad14f22aec571f/r/tests/testthat/test-dplyr-funcs-datetime.R#L2027) for the situation when `"I"` is greater than 12.